Prepare release 1.7.0 (#509)
exasol/python-toolbox: doc/user_guide/features/creating_a_release.rst

Creating a release
==================

Preparing a release
+++++++++++++++++++

The ``release:prepare`` nox session affects files in the ``doc/changes`` directory:

* Creates & switches to a release branch (can be skipped with ``--no-branch``)

> * Updates the version in the ``pyproject.toml``

Author: No, these two changes still happen. Good wording/consistency catch..

* Moves the content of unreleased changes file ``unreleased.md`` to a versioned changes file ``changes_<version>.md``
* Appends to the versioned changes file any direct dependency changes between the current ``poetry.lock`` and the one from the latest tag
* Updates the ``changelog.md`` list with the newly create versioned changes file
* Commits the changes (can be skipped with ``--no-add``)
* Pushes the changes and creates a PR (can be deactivated with ``--no-pr``)

After a PR is created, approved, & merged into the default branch. A developer can use
the ``release:trigger`` nox session which:

* Switches & pulls the changes from the default branch
* Verifies that the version to be released does not already have a git tag or GitHub release

> * Creates a new tag & pushes it to the default branch, which will trigger the GitHub workflow ``cd.yml``

Reviewer: ```suggestion
* Create a new tag & push it to the default branch, which will trigger the GitHub workflow ``cd.yml``
```

> Simple instructions

Reviewer: ```suggestion
Triggering the Release
```

-------------------

The ``release:trigger`` nox session

> #. Prepare the project for a new release:
> 
>     .. code-block:: shell
> 
>          nox -s release:prepare -- --type {major,minor,patch}
> 
> #. Merge your **Pull Request** to the **default branch**
> 
> #. Trigger the release:
> 

Reviewer: Probably redundant, by now.

```suggestion
```
